app/coffee/AuthHandler.coffee

```diff
@@ -0,0 +1,85 @@
+_ = require("lodash")
+bcrypt = require("bcrypt")
+crypto = require("crypto")
+logger = require("logger-sharelatex")
+
+{ db } = require("./MongoHandler")
+Errors = require("./Errors")
+
+TOKEN_TTL_MS = 60 * 60 * 1000    # 1 hour
+TOKEN_SEARCH_PREFIX_LENGTH = 16  # 8 bytes
```

Comment:
The collision probability seems [low enough](http://davidjohnstone.net/pages/hash-collision-probability), so this is probably long enough.

Reply:
:) yes. and this is only among tokens for a specific email, of which there is reasonably only a handful.